Lib/importlib/abc.py

```diff
@@ -15,20 +15,36 @@
 import abc
 import warnings
 
-# for compatibility with Python 3.10
-from .resources.abc import ResourceReader, Traversable, TraversableResources
+from .resources import abc as _resources_abc
 
 
 __all__ = [
     'Loader', 'Finder', 'MetaPathFinder', 'PathEntryFinder',
     'ResourceLoader', 'InspectLoader', 'ExecutionLoader',
     'FileLoader', 'SourceLoader',
-
-    # for compatibility with Python 3.10
-    'ResourceReader', 'Traversable', 'TraversableResources',
 ]
 
 
+def __getattr__(name, canonical=_resources_abc):
+    """
+    For backwards compatibility, continue to make names
+    from canonical available through this module.
+    """
+    if name in canonical.__all__:
+        obj = getattr(canonical, name)
+        import warnings
+        warnings.warn(
+            f"Using or importing the ABCs from {__name__!r} instead "
+            f"of from {canonical.__name__!r} is deprecated since "
+            "Python 3.11, and in 3.13 it will stop working",
```

Comment:
nitpic: this is the old wording from the collections.abc deprecation/removal but the wording "now deprecated, scheduled for remove in Python 3.XX` is more common:

```suggestion
            f"of from {canonical.__name__!r} is now deprecated, "
            "scheduled for removal in Python 3.13",
```

Reply:
Agreed. That's better.